### PR TITLE
Make m5 the default instance class

### PR DESCRIPTION
**Pull request by aaw**
If a terraform config doesn't explicitly provide a container profile for an app or database, we fill in m4. Since we're going to upgrade everyone to m5 starting a week from now, we need to default to m5 instead. Otherwise, after we upgrade everyone, a terraform config that doesn't explicitly provide a container profile will generate a plan to downgrade everything to m4. 

Running this provider against your config before your apps and databases have been upgraded will just result in an early upgrade to m5 for apps and databases, which can be overridden by explicitly specifying m4 if the upgrade is unwanted.

This PR couples with https://github.com/aptible/go-deploy/pull/64, I've released that and bumped the version here with just `go get github.com/aptible/go-deploy`

### CODE REVIEW

**Comment by aaw** [comment]
https://app.shortcut.com/aptible/story/17597/support-m5-as-default

**Comment by aaw** [comment]
@joshraker friendly ping for a review

**Review by joshraker — CHANGES_REQUESTED** [review]
Just one oddity in the `go.sum`. If you're up for a challenge you could update the profile validation to raise a warning, rather than an error so that the next time we add a profile users can use it without having to update their client right away.

File: go.sum [hunk 1]
```diff
@@ -117,6 +117,8 @@ github.com/apparentlymart/go-textseg/v13 v13.0.0 h1:Y+KvPE1NYz0xl601PVImeQfFyEy6
 github.com/apparentlymart/go-textseg/v13 v13.0.0/go.mod h1:ZK2fH7c4NqDTLtiYLvIkEghdlcqw7yxLeM89kiTRPUo=
 github.com/aptible/go-deploy v0.5.0 h1:fVqCoyFm3oGifyUyyoY9dJUxW3bwhgu9C2oTREXCiSI=
 github.com/aptible/go-deploy v0.5.0/go.mod h1:h0Zt8I+pV3aqvPo+rA1hIeQjT/13RzFPYuTmg4+SyPQ=
```
[the remark this answers is not in the document]
Comment (aaw) [reply go.sum]: Thanks! `go mod tidy` cleaned it up

**Comment by aaw** [comment]
> Just one oddity in the `go.sum`. If you're up for a challenge you could update the profile validation to raise a warning, rather than an error so that the next time we add a profile users can use it without having to update their client right away.

Turns out the validations for instance profile are [already warnings](https://github.com/aptible/terraform-provider-aptible/blob/master/aptible/validation.go#L79). I tested an example with a bogus instance profile anyway and saw `terraform plan` only give me a warning.

**Review by joshraker — APPROVED** [review]

**Pull request merged**